iOS/MMIWG2S/Views/ARFaceViewerUI.swift

```diff
@@ -107,4 +167,56 @@ class ARFaceViewerUI: UIView {
         redSelected = true
         delegate?.textureChanged(toRed: true)
     }
+
+    @objc private func shareButtonTapped() {
+        generateLightFeedback()
+        backButton?.isHidden = true
+        shareButton?.isHidden = true
+
+        // Add a pause to allow for previous views to disappear.
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.15) { [weak self] in
+            self?.shareImage()
+        }
+    }
+
+    @objc private func backButtonTapped() {
+        generateLightFeedback()
+        delegate?.resumeSessionOnceCaptured()
+        changeViewVisibility(isCapturing: true)
+    }
+
+    private func generateLightFeedback() {
+        let impactFeedbackgenerator = UIImpactFeedbackGenerator(style: .light)
+        impactFeedbackgenerator.prepare()
```

Comment:
This doesn't seem like its doing anything. According to [Apple's documentation](https://developer.apple.com/documentation/uikit/uifeedbackgenerator#2555405) prepare should not be called directly before 1impactOccurred`, otherwise the haptic engine has not been given enough time to make any noticeable reduction in latency. 

I think what you want to do is add a second function `prepareForLightFeedback()` which would be called in an `@objc fun buttonWillTap`. This would get added as a target for all of the buttons it's used in on a `.touchDown` event. You'll just need to remember to also set `impactFeedbackgenerator = nil` after each impact _and_ after each button would send a `.touchCancel` event (in order to avoid memory leaks).

Reply:
Thanks for the docs. The blogs mentioned nothing about all this... This docs list 4 things to do
```
To use a feedback generator, the following are required:
Instantiating the Generator
Preparing the Generator (optional)
Triggering Feedback
Releasing the Generator (optional).
```
I saw preparing was optional and I read a little more. They say you only really need to prepare if you are trying to time it with sound or video. Which we're not. So I just removed it, as you were right, it's not doing anything. I did go through the step of deallocating though. 

I hope that's good :)

Reply:
Perfecto! Ya deallocating shouldn't be necessary if we're not calling prepare since it can stay a local var